pkg/assets/builder.go

```diff
@@ -146,6 +147,20 @@ func (a *AssetBuilder) RemapImage(image string) (string, error) {
 		}
 	}
 
+	if a.AssetsLocation != nil && a.AssetsLocation.ContainerProxy != nil {
+		containerProxy := *a.AssetsLocation.ContainerProxy
+		normalized := image
+
+		// If the image name contains only a single / we can assume ot is an image pulled from the docker hub. eg. "weaveworks/weave-kube"
+		// In these cases it should be sufficient to just prepend the proxy url, producing eg docker-proxy/weaveworks/weave-kube
+		if strings.Count(normalized, "/") == 1 {
+			normalized = containerProxy + "/" + normalized
+		} else {
+			var re = regexp.MustCompile(`^[^/]+`)
+			normalized = re.ReplaceAllString(normalized, containerProxy)
```

Comment:
I think this is right, but I would definitely write a simple unit test when there's a regex involved!

I'm guessing if you had your mirror proxy as `mirror.example.com:5000` that e.g. `gcr.io/kubernetes/api:1.1` would go to `mirror.example.com:5000/kubernetes/api:1.1` - that seems right

One challenge is that we recently introduced a vanity URL, so the image names for some kubernetes components is now `k8s.gcr.io/kube-proxy`.  I don't know how you want to rewrite that, but I'm guessing it will need to end up as `mirror.example.com:5000/kubernetes/kube-proxy`.  I think it's fine to special case `k8s.gcr.io` if you have to.

The alternative to special casing it would be accept `mirror.example.com:5000/k8s.gcr.io/kube-proxy`.  I think that's actually what your code does today.  But it's subtle, and so I think having a few unit tests with comments about why things work the way they do would be very helpful.

For a unit test, I suggest just splitting this out into a small function if you have any trouble with setting up an AssetBuilder.  (But if you don't - no problem!)

Also a question for you - what are you using as your pull-through cache?